src/uct/cuda/cuda_ipc/cuda_ipc_cache.c

```diff
@@ -183,230 +139,165 @@ uct_cuda_ipc_get_remote_cache(pid_t pid, uct_cuda_ipc_cache_t **cache)
+
+    uct_cuda_ipc_close_memhandle(cuda_ipc_region->mapping_start);
+
+}
+
+static void uct_cuda_ipc_rcache_dump_region(void *context, ucs_rcache_t *rcache,
+                                            ucs_rcache_region_t *region, char *buf,
+                                            size_t max)
+{
+    uct_cuda_ipc_rcache_region_t *cuda_ipc_region =
+                    ucs_derived_of(region, uct_cuda_ipc_rcache_region_t);
+
+    snprintf(buf, max, "mapping_start %p", cuda_ipc_region->mapping_start);
+}
+
+static ucs_rcache_ops_t uct_cuda_ipc_rcache_ops = {
+    .mem_reg     = uct_cuda_ipc_rcache_mem_reg,
+    .mem_dereg   = uct_cuda_ipc_rcache_mem_dereg,
+    .dump_region = uct_cuda_ipc_rcache_dump_region
+};
+
+
+ucs_status_t
+uct_cuda_ipc_create_cache(uct_cuda_ipc_md_t *md, ucs_rcache_t **cache,
+                          const char *name)
 {
     ucs_status_t status;
-    uct_cuda_ipc_cache_t *cache_desc;
-    int ret;
+    ucs_rcache_t *cache_desc;
+    ucs_rcache_params_t rcache_params;
 
-    cache_desc = ucs_malloc(sizeof(uct_cuda_ipc_cache_t), "uct_cuda_ipc_cache_t");
+    cache_desc = ucs_malloc(sizeof(struct ucs_rcache), "ucs_rcache");
     if (cache_desc == NULL) {
-        ucs_error("failed to allocate memory for cuda_ipc cache");
+        ucs_error("failed to allocate memory for cuda_ipc rcache");
         return UCS_ERR_NO_MEMORY;
     }
 
-    ret = pthread_rwlock_init(&cache_desc->lock, NULL);
-    if (ret) {
-        ucs_error("pthread_rwlock_init() failed: %m");
-        status = UCS_ERR_INVALID_PARAM;
-        goto err;
-    }
-
-    status = ucs_pgtable_init(&cache_desc->pgtable,
-                              uct_cuda_ipc_cache_pgt_dir_alloc,
-                              uct_cuda_ipc_cache_pgt_dir_release);
+    rcache_params.region_struct_size = sizeof(uct_cuda_ipc_rcache_region_t);
+    rcache_params.ucm_events         = 0;
+    rcache_params.ucm_event_priority = 0;
+    rcache_params.ops                = &uct_cuda_ipc_rcache_ops;
+    rcache_params.context            = NULL;
+    rcache_params.flags              = 0;
+    rcache_params.max_regions        = md->rcache_max_regions;
+    rcache_params.max_size           = md->rcache_max_size;
+
+    status = ucs_rcache_create(&rcache_params, name,
+                               ucs_stats_get_root(), &cache_desc);
     if (status != UCS_OK) {
-        goto err_destroy_rwlock;
-    }
-
-    cache_desc->name = strdup(name);
-    if (cache_desc->name == NULL) {
-        status = UCS_ERR_NO_MEMORY;
-        goto err_destroy_rwlock;
+        ucs_error("failed to create cuda_ipc remote cache: %s",
+                  ucs_status_string(status));
+        goto err;
     }
 
     *cache = cache_desc;
     return UCS_OK;
 
-err_destroy_rwlock:
-    pthread_rwlock_destroy(&cache_desc->lock);
 err:
     free(cache_desc);
     return status;
 }
 
-void uct_cuda_ipc_destroy_cache(uct_cuda_ipc_cache_t *cache)
+void uct_cuda_ipc_destroy_cache(ucs_rcache_t *cache)
```

Comment:
Maybe remove the wrapper. And just call `ucs_rcache_destroy` in the static cleanup method.